### PR TITLE
[Communication] - All - Add Identifier suffix to all CommunicationIdentifiers

### PR DESCRIPTION
**Pull request by minnieliu**

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 package com.azure.communication.chat.models;
 
-import com.azure.communication.common.CommunicationUser;
+import com.azure.communication.common.CommunicationUserIdentifier;
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java:7]: This is generated code. Is this the result of swagger file change and regeneration or a manual change?

File: sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java [hunk 2]
```diff
@@ -219,8 +219,8 @@ public void listChatThreadMember() {
     public void addChatThreadMembers() {
         ChatThreadClient chatThreadClient = getChatThreadClient();
 
-        CommunicationUser user1 = new CommunicationUser("Id 1");
-        CommunicationUser user2 = new CommunicationUser("Id 2");
+        CommunicationUserIdentifier user1 = new CommunicationUserIdentifier("Id 1");
+        CommunicationUserIdentifier user2 = new CommunicationUserIdentifier("Id 2");
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java:224]: Which README.md has reference to this code? Anything in that README.md should be changed accordingly?

**Review by JoshuaLai — APPROVED** [review]
Just simple name changes

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/PhoneNumberIdentifier.java [hunk 3]
```diff
@@ -18,7 +18,7 @@
      * E.164 is a phone number formatted as +[CountryCode][AreaCode][LocalNumber] eg. "+18005555555"
```
Comment (DominikMe) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/PhoneNumberIdentifier.java:18]: Creates a PhoneNumber object => Creates a PhoneNumberIdentifier object

Probably worthwhile to look for similar comments in the other classes too. "the string identifier representing the PhoneNumber" can stay unchanged, just add the *Identifier when referring to the exact type.

**Review by DominikMe — APPROVED** [review]
Lgtm! One comment to follow up on JavaDoc fixes where needed.

**Comment by JianpingChen** [comment]
Is the underlying swagger file changed? If so, should that be updated? Also some of the automated tests are changed. Should the saved replay json file be updated?

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationUserIdentifier.java [hunk 4]
```diff
@@ -17,7 +17,7 @@
      * @param id the string identifier representing the identity
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationUserIdentifier.java:17]: Creates a _CommunicationUserIdentifier_ object

**Comment by JianpingChen** [comment]
@juancamilor FYI: When this PR is merged into the master, your branch for Preview3 may need to be rebased from master to pick up this change.

**Review by jbeauregardb — APPROVED** [review]
LGTM, just left one comment :)

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java:7]: @JianpingChen This is actually not generated code, we're just adding these new classes. All generated code is in the implementation folder.

File: sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java (hunk 2 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java:224]: No, these lines are not used directly in the README just there for complication purposes.

**Comment by minnieliu** [comment]
> Is the underlying swagger file changed? If so, should that be updated? Also some of the automated tests are changed. Should the saved replay json files be updated?

The swagger file is not changed

**Comment by JianpingChen** [comment]
CallingApplication also inherits from CommunicationIdentifier. Should that be renamed as well?

public class CallingApplication extends CommunicationIdentifier

**Comment by juancamilor** [comment]
@minnieliu / all, I wonder if it may be better to do all these changes in the current feature branch instead so the merging for GA (or before) get easier ? (@DominikMe  @JianpingChen  @danielgerlag )

**Comment by minnieliu** [comment]
> CallingApplication also inherits from CommunicationIdentifier. Should that be renamed as well?
> 
> public class CallingApplication extends CommunicationIdentifier

JP, it is already renamed :): 
image

**Comment by JianpingChen** [comment]
> > CallingApplication also inherits from CommunicationIdentifier. Should that be renamed as well?
> > public class CallingApplication extends CommunicationIdentifier
> 
> JP, it is already renamed :):
> image

Great! Sorry for missing that. I was scanning through "jump to" file list and missed it.

As Camilo is suggesting in SDK channel, it is probably better to merge this into AzureCommunication feature branch.

**Comment by DominikMe** [comment]
@juancamilor master is the source of truth. Feature branches are short-lived and need to rebase on master to get the latest changes. Happy to chat more.

**Pull request merged**